microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/KafkaPublisher.java

```diff
@@ -0,0 +1,311 @@
+                emiter.fail(e);
+            } finally {
+                taskLock.unlock();
+            }
+        }, 0, periodExecutions, TimeUnit.MILLISECONDS);
+    }
+
+    private int currentNoAck() {
+        return pendingCommits.values().stream().map(list -> list.size()).reduce((a, b) -> a + b).orElse(0);
+    }
+
+    /**
+     * Process the ACKs only if enable.auto.commit is false.
+     * This will search events that are ACK and it will commit them to Kafka.
+     * What ever the commit was success of not, it will be notified to the message.
+     */
+    private void processACK() {
+        Map<TopicPartition, OffsetAndMetadata> offsets = new LinkedHashMap<>();
+        List<KafkaMessage<K, V>> notifications = new LinkedList<>();
+        // Commit highest offset + 1 of each partition that was ACK, and remove from pending
+        for (Entry<TopicPartition, List<KafkaMessage<K, V>>> entry : pendingCommits.entrySet()) {
+            // No need to sort it, offsets are consumed in order
+            List<KafkaMessage<K, V>> byPartition = entry.getValue();
+            Iterator<KafkaMessage<K, V>> iterator = byPartition.iterator();
+            KafkaMessage<K, V> highest = null;
+            while (iterator.hasNext()) {
+                KafkaMessage<K, V> element = iterator.next();
+                if (element.isAck()) {
+                    notifications.add(element);
+                    highest = element;
+                    iterator.remove();
+                } else {
+                    break;
+                }
+            }
+            if (highest != null) {
+                OffsetAndMetadata offset = new OffsetAndMetadata(highest.getPayload().offset() + 1);
+                LOGGER.fine(() -> String.format("Will commit %s %s", entry.getKey(), offset));
+                offsets.put(entry.getKey(), offset);
+            }
+        }
+        if (!notifications.isEmpty()) {
+            Optional<RuntimeException> exception = commitInKafka(offsets);
+            notifications.stream().forEach(message -> {
+                exception.ifPresent(ex -> message.exception(ex));
+                message.wakeUp();
+            });
+        }
+    }
+
+    private Optional<RuntimeException> commitInKafka(Map<TopicPartition, OffsetAndMetadata> offsets) {
+        LOGGER.fine(() -> String.format("%s events to commit: ", offsets.size()));
+        LOGGER.fine(() -> String.format("%s", offsets));
+        try {
+            kafkaConsumer.commitSync(offsets);
+            LOGGER.fine(() -> "The commit was successful");
+            return Optional.empty();
+        } catch (RuntimeException e) {
+            LOGGER.log(Level.SEVERE, "Unable to commit in Kafka " + offsets, e);
+            return Optional.of(e);
+        }
+    }
+
+    /**
+     * Closes the connections to Kafka and stops to process new events.
+     */
+    @Override
+    public void close() {
+        // Stops pooling
+        kafkaConsumer.wakeup();
```

Comment:
Wakeup should normally get you from polling loop, but in our case its inside the scheduled runnable, It can be better to stop scheduler before wakeup and in the catch block jump out:
```java
	                    } catch (WakeupException e) {
                                LOGGER.fine(() -> "It was requested to stop polling from channel");
                                return;
                            }
```

Reply:
You already stopped it in connector, sorry